xarray/core/dtypes.py

```diff
@@ -34,6 +34,8 @@ def maybe_promote(dtype):
         fill_value = np.datetime64('NaT')
     elif np.issubdtype(dtype, np.timedelta64):
         fill_value = np.timedelta64('NaT')
+    elif dtype.kind == 'b':
+        fill_value = False
```

Comment:
This is convenient for me, but it is not very clear whether `False` is equivalent to `nan` for boolean arrays.
If anyone has objections, I will consider different approach.

Reply:
Indeed, let's consider other options here. This is used for the default value when reindexing/aligning.